src/sentry/web/frontend/react_page.py

```diff
@@ -79,6 +96,21 @@ def handle_react(self, request: Request, **kwargs) -> HttpResponse:
             )
             return HttpResponseRedirect(redirect_url)
 
+        # We don't allow HTML pages to be served from region domains.
+        if request.subdomain and subdomain_is_region(request):
+            redirect_url = resolve_activeorg_redirect_url(request)
+            if redirect_url:
+                logger.info(
+                    "react_page.redirect.regiondomain",
+                    extra={"path": request.path, "location": redirect_url},
+                )
+                return HttpResponseRedirect(redirect_url)
+            else:
+                raise Http404()
```

Comment:
Wouldn't it be better to redirect to `system.url-prefix`? Or does that cause issues?

Reply:
I was worried about a redirect loop. As going to `/` goes back into `ReactPage` which contains many redirects.  Going to `us.sentry.io` is generally a mistake as customers should be using their org slugs as subdomains.